### PR TITLE
Document the Usage of the appium-flutter-integration-driver

### PR DESCRIPTION
**Pull request by bahrimootaz**
### Description

Announcing to customers that they can run tests on Sauce Labs using flutter apps and appium-flutter-integration-driver. 

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

**Review by kristofmuhi — APPROVED** [review]

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+title: Appium Flutter Integration Driver Testing on Sauce Labs
+sidebar_label: Appium Flutter Integration Testing
+description: Learn how to test Flutter apps with the Appium Flutter Integration Driver.
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Sauce Labs supports testing Flutter apps on Android and iOS virtual and real devices with Appium by utilizing the [appium-flutter-integration-driver](https://github.com/AppiumTestDistribution/appium-flutter-integration-driver).
```
Comment (kristofmuhi) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:12]: @bahrimootaz remove virtual. only RDC supported

**Review by kristofmuhi — CHANGES_REQUESTED** [review]

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 1 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:12]: done

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+        appiumVersion: 'appium-20240701'
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+</TabItem>
+<TabItem value="python">
```
Comment (kristofmuhi) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:520]: java and wdio client bindings support only. can you remove everything that is not java/js binding?

**Review by kristofmuhi — CHANGES_REQUESTED** [review]

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+| Running integration tests for Flutter apps with embedded webview or native view, or existing native apps with embedded Flutter view | ❌                    | ✔️                                |
+| Running tests on multiple devices simultaneously                                                                                    | ❌                    | ✔️                                |
+| Running integration tests on device farms that offer Appium support                                                                 | ❌                    | ✔️                                |
+| App interactions beyond Flutter’s contextuality (e.g., sending an OTP from a message application)                                   | ❌                    | ✔️                                |
+
+## Differences from Appium Flutter Driver
+
+The current Appium Flutter Driver is built on top of the `flutter_test` SDK, which is deprecated. The potential deprecation ([Expand deprecation policy to package:flutter_driver](https://github.com/flutter/flutter/issues/139249)) means this driver may not work with future Flutter updates. It also does not handle all cases, such as permission dialog handling.
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:30]: This bit `Expand deprecation policy to package:flutter_driver` looks strange. 

image

Comment (diemol) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:30]: Did you mean just to link the word `deprecation`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+## What You'll Need
+
+- Familiarity with creating, signing and building [Flutter apps](https://docs.flutter.dev/).
+- Familiarity writing and running [Appium tests](/mobile-apps/automated-testing/appium/).
+
+## Use Sauce Labs `My Demo App Flutter`
+
+[*My Demo App Flutter*](https://github.com/saucelabs/my-demo-app-flutter) is a mobile application developed using Flutter based
+on [Flutter Counter example application](https://github.com/felangel/bloc/tree/master/examples/flutter_counter).
+Modified by the Sauce Labs team, this app is designed to demonstrate the robust capabilities of Sauce Labs'
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:58]:
```suggestion
Modified by the Sauce Labs team, this app is designed to demonstrate the robust capabilities of Sauce Labs
```

Comment (diemol) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:58]: I don't remember the exact explanation, but Sauce Labs does not need a `'` for possessive.

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+
+You can now upload the built apps with our [REST API](/dev/api/storage/#upload-file-to-app-storage), or manually upload them to the preferred Data Center. See [Manually Uploading an App](/mobile-apps/live-testing/live-mobile-app-testing/#uploading-an-app) for more information.
+
+## Configuring your Appium Capabilities
+
+More information on how to write Appium tests for Flutter apps can be found in the [appium-flutter-integration-driver](https://github.com/AppiumTestDistribution/appium-flutter-integration-driver) repository. You can also find a sample Flutter app and tests in the [Demo JS - Appium Flutter Integration](https://github.com/saucelabs-training/demo-js/tree/main/webdriverio/appium-app/examples/appium-flutter-integration)-repository.
+
+:::info W3C Capabilities
+We encourage you to use W3C capabilities for your tests. For more information on W3C capabilities, see [W3C Capabilities](/dev/test-configuration-options/#mobile-appium-capabilities).
+:::
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:139]: Is this needed? Appium 2 only supports W3C anyway.

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 6]
```diff
@@ -0,0 +1,678 @@
...
+
+capabilities.setCapability("platformName", "android");
+capabilities.setCapability("appium:platformVersion", "12");
+capabilities.setCapability("appium:deviceName", "Google Pixel Emulator");
+// Mandatory for using the appium-flutter-integration-driver
+capabilities.setCapability("appium:automationName", "FlutterIntegration");
+capabilities.setCapability("appium:app", "storage:filename=sl_my_demo_flutter_app.apk");
+
+HashMap<String, Object> sauceOptions = new HashMap<String, Object>();
+sauceOptions.put("appiumVersion", "appium2-20240701");
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 7]
```diff
@@ -0,0 +1,678 @@
...
+
+capabilities.setCapability("platformName", "ios");
+capabilities.setCapability("appium:platformVersion", "15.4");
+capabilities.setCapability("appium:deviceName", "iPhone 13 Simulator");
+// Mandatory for using the appium-flutter-integration-driver
+capabilities.setCapability("appium:automationName", "FlutterIntegration");
+capabilities.setCapability("appium:app", "storage:filename=sl_my_demo_flutter_app.ipa");
+
+HashMap<String, Object> sauceOptions = new HashMap<String, Object>();
+sauceOptions.put("appiumVersion", "appium2-20240701");
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

**Review by diemol — CHANGES_REQUESTED** [review]

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+capabilities.setCapability("appium:platformVersion", "12");
+capabilities.setCapability("appium:deviceName", "Google Pixel 6");
+// Mandatory for Appium 2
+capabilities.setCapability("appium:automationName", "FlutterIntegration");
+capabilities.setCapability("appium:app", "storage:filename=sl_my_demo_flutter_app.apk");
+
+HashMap<String, Object> sauceOptions = new HashMap<String, Object>();
+// appiumVersion is mandatory to use Appium 2 on Sauce Labs
+sauceOptions.put("appiumVersion", "appium-20240701");
+capabilities.setCapability("sauce:options", sauceOptions);
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:183]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+capabilities.setCapability("appium:platformVersion", "16");
+capabilities.setCapability("appium:deviceName", "iPhone 14");
+// Mandatory for Appium 2
+capabilities.setCapability("appium:automationName", "FlutterIntegration");
+capabilities.setCapability("appium:app", "storage:filename=sl_my_demo_flutter_app.ipa");
+
+HashMap<String, Object> sauceOptions = new HashMap<String, Object>();
+// appiumVersion is mandatory to use Appium 2 on Sauce Labs
+sauceOptions.put("appiumVersion", "appium-20240701");
+capabilities.setCapability("sauce:options", sauceOptions);
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:206]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+    platformName: 'android',
+    // W3C Protocol is mandatory for Appium 2
+    'appium:platformVersion': '12',
+    'appium:deviceName': 'Google Pixel 6',
+    // Mandatory for Appium 2
+    'appium:automationName': 'FlutterIntegration',
+    'appium:app': 'storage:filename=sl_my_demo_flutter_app.apk',
+    'sauce:options': {
+        // appiumVersion is mandatory to use Appium 2 on Sauce Labs
+        appiumVersion: 'appium-20240701'
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:235]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+    platformName: 'ios',
+    // W3C Protocol is mandatory for Appium 2
+    'appium:platformVersion': '16',
+    'appium:deviceName': 'iPhone 14',
+    // Mandatory for Appium 2
+    'appium:automationName': 'FlutterIntegration',
+    'appium:app': 'storage:filename=sl_my_demo_flutter_app.ipa',
+    'sauce:options': {
+        // appiumVersion is mandatory to use Appium 2 on Sauce Labs
+        appiumVersion: 'appium-20240701'
```
Comment (diemol) [inline docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:255]: What about `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` inside `sauceOptions`?

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 7 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md]: done

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 6 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md]: done

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 5 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:139]: true! will remove it now

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 4 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:58]: removed

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 2 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md]: done

File: docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md (hunk 3 above)
Comment (bahrimootaz) [reply docs/mobile-apps/automated-testing/appium/appium-flutter-integration-driver.md:30]: no. that is correct. 
Source of this is here : https://github.com/AppiumTestDistribution/appium-flutter-integration-driver

**Review by kristofmuhi — APPROVED** [review]

**Pull request merged**